packages/expo-updates/android/src/main/java/expo/modules/updates/statemachine/UpdatesStateContext.kt

```diff
@@ -86,6 +93,18 @@ data class UpdatesStateContext(
           }
           putBundle("downloadError", errorMap)
         }
+        if (lastCheckForUpdateTime != null) {
+          putString("lastCheckForUpdateTimeString", DATE_FORMATTER.format(lastCheckForUpdateTime))
+        }
       }
     }
+
+  companion object {
+    val DATE_FORMATTER: SimpleDateFormat
+      get() {
+        val formatter = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.US)
+        formatter.timeZone = java.util.TimeZone.getTimeZone("GMT")
+        return formatter
+      }
```

Comment:
getters will still run on each invocation I believe. I think you want is a `lazy` val?

```suggestion
    val DATE_FORMATTER: SimpleDateFormat by lazy {
      SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.US).apply {
        timeZone = java.util.TimeZone.getTimeZone("GMT")
      }
    }
```

Reply:
Aha so that's how you do that in Kotlin :)